binary-tagged-0.1.4.1 failed to rebuild in Stackage Nightly

Maybe some new dependency conflict?

```
src/Data/Binary/Tagged.hs:324:59: error:
    Ambiguous occurrence ‘datatypeName’
    It could refer to either ‘SOP.datatypeName’,
                             imported from ‘Generics.SOP’ at src/Data/Binary/Tagged.hs:104:1-36
                             (and originally defined in ‘Generics.SOP.Metadata’)
                          or ‘Data.Binary.Tagged.datatypeName’,
                             defined at src/Data/Binary/Tagged.hs:358:1

src/Data/Binary/Tagged.hs:325:59: error:
    Ambiguous occurrence ‘datatypeName’
    It could refer to either ‘SOP.datatypeName’,
                             imported from ‘Generics.SOP’ at src/Data/Binary/Tagged.hs:104:1-36
                             (and originally defined in ‘Generics.SOP.Metadata’)
                          or ‘Data.Binary.Tagged.datatypeName’,
                             defined at src/Data/Binary/Tagged.hs:358:1

src/Data/Binary/Tagged.hs:345:35: error:
    Ambiguous occurrence ‘datatypeName’
    It could refer to either ‘SOP.datatypeName’,
                             imported from ‘Generics.SOP’ at src/Data/Binary/Tagged.hs:104:1-36
                             (and originally defined in ‘Generics.SOP.Metadata’)
                          or ‘Data.Binary.Tagged.datatypeName’,
                             defined at src/Data/Binary/Tagged.hs:358:1

src/Data/Binary/Tagged.hs:351:47: error:
    Ambiguous occurrence ‘datatypeName’
    It could refer to either ‘SOP.datatypeName’,
                             imported from ‘Generics.SOP’ at src/Data/Binary/Tagged.hs:104:1-36
                             (and originally defined in ‘Generics.SOP.Metadata’)
                          or ‘Data.Binary.Tagged.datatypeName’,
                             defined at src/Data/Binary/Tagged.hs:358:1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,5 @@
+- 0.1.4.2 (2016-12-05)
+    - Fix compilation with generics-sop-0.2.3.0
 - 0.1.4.1 (2016-09-20)
     - Print mismatching hashes in base16 encoding
 - 0.1.4.0 (2016-04-12)

--- a/package.yaml
+++ b/package.yaml
@@ -1,5 +1,5 @@
 name: binary-tagged
-version: 0.1.4.1
+version: 0.1.4.2
 synopsis: Tagged binary serialisation.
 description: Check <https://github.com/phadej/binary-tagged#readme README on Github>
 author: Oleg Grenrus <oleg.grenrus@iki.fi>

--- a/src/Data/Binary/Tagged.hs
+++ b/src/Data/Binary/Tagged.hs
@@ -354,9 +354,11 @@ sopStructuralInfo1S nsop di = NominalNewtype (datatypeName di) nsop
 -- SOP helpers
 -------------------------------------------------------------------------------
 
+#if !MIN_VERSION_generics_sop(0,2,3)
 datatypeName :: DatatypeInfo xss -> DatatypeName
 datatypeName (Newtype _ d _)  = d
 datatypeName (ADT _ d _)      = d
+#endif
 
 -- | Interleaving
 --

